internal/node/node.go

```diff
@@ -91,16 +92,61 @@ func (dn *Node) SpawnDescendant() *Node {
 // setters
 // ---------------------------------------------------------------------------
 
+type addAttrConfig struct {
+	doNotAddToSpan       bool
+	addToOTELHTTPLabeler bool
+	otelHTTPLabeler      otelHTTPLabeler
+}
+
+type addAttrOptions func(*addAttrConfig)
+
+func makeAddAttrConfig(opts ...addAttrOptions) addAttrConfig {
+	cfg := addAttrConfig{}
+
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	return cfg
+}
+
+func DoNotAddToSpan() addAttrOptions {
+	return func(cfg *addAttrConfig) {
+		cfg.doNotAddToSpan = true
+	}
+}
+
+func AddToOTELHTTPLabeler(ctx context.Context) addAttrOptions {
+	return func(cfg *addAttrConfig) {
+		cfg.addToOTELHTTPLabeler = true
+
+		// cannot be nil, by contract of the otelhttp package.
+		cfg.otelHTTPLabeler, _ = otelhttp.LabelerFromContext(ctx)
```

Comment:
have you tested this updated version of clues with code that attempts to output spans? The documentation for `otelhttp` has the following for [LabelerFromContext](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp#LabelerFromContext)

> If no Labeler was found in the provided context a new, empty Labeler is returned and the second return value is false. In this case it is safe to use the Labeler but any attributes added to it will not be used.

That makes it seem like a corresponding call to [ContextWithLabeler](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp#ContextWithLabeler) may be required in some cases

Reply:
hah, I completely missed that.  Thanks for the catch, I'll get this tidied up.